### PR TITLE
chore: bump hyper to 1.0.1

### PR DESCRIPTION
**Pull request by mraszyk**
This PR bumps hyper to 1.0.1.

### CODE REVIEW

**Comment by adamspofford-dfinity** [comment]
Is this prompted by something? Otherwise I would prefer to keep 0.14 until reqwest updates to hyper 1.0, so there is only one copy in the tree.

**Comment by mraszyk** [comment]
> Is this prompted by something? Otherwise I would prefer to keep 0.14 until reqwest updates to hyper 1.0, so there is only one copy in the tree.

This blocks bumping axum to 0.7 in icx-proxy.

File: ic-agent/Cargo.toml [hunk 1]
```diff
@@ -60,11 +60,15 @@ version = "2.0.1"
 optional = true
 
 [target.'cfg(not(target_family = "wasm"))'.dependencies]
-hyper-rustls = { version = "0.24.1", features = [
+http-body-to-bytes = "0.2.0"
+http-body-util = "0.1.0"
+hyper-util = { version = "0.1.3", features = ["client", "client-legacy", "http2"] }
```
[the remark this answers is not in the document]
Comment (mraszyk) [reply ic-agent/Cargo.toml:65]: What do you mean?

[the remark this answers is not in the document]
Comment (mraszyk) [reply ic-agent/Cargo.toml]: I've added `optional = true` to all of them if that's what you mean.

**Review by adamspofford-dfinity — CHANGES_REQUESTED** [review]
This code does not build with `--features hyper`.

File: ic-agent/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamspofford-dfinity) [reply ic-agent/Cargo.toml]: As in, they should only be enabled when the hyper feature is enabled.

[the remark this answers is not in the document]
Comment (mraszyk) [reply ic-agent/Cargo.toml]: Is it ok now?

**Comment by mraszyk** [comment]
> This code does not build with `--features hyper`.

What do you mean? The CI pipeline has succeeded and I could also successfully run `cargo build --release --features hyper` locally.

**Comment by adamspofford-dfinity** [comment]
Apologies for not adding detail - the new errors look like this:
```
error[E0432]: unresolved import `http_body_to_bytes`
 --> ic-agent/src/agent/http_transport/hyper_transport.rs:7:5
  |
7 | use http_body_to_bytes::{http_body_to_bytes, http_body_to_bytes_with_max_length};
  |     ^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `http_body_to_bytes`
```
Because the optional dependencies were not added to the hyper feature. I am not sure why it is succeeding on your local machine. I am fixing the CI to detect this in #528.

**Comment by mraszyk** [comment]
I could reproduce the build failure and fixed it now.

**Review by adamspofford-dfinity — APPROVED** [review]

**Pull request merged**